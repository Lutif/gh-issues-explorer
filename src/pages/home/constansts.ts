export const orgs = [
  "reactjs/reactjs.org",
  "mui-org/material-ui",
  "FormidableLabs/nuka-carousel",
];
