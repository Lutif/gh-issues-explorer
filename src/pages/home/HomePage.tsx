import React from "react";
import { useHistory } from "react-router-dom";
import { orgs } from "./constansts";
import { LinksWrapper } from "./homepage.styles";

interface HomePageProps {}

export const HomePage: React.FC<HomePageProps> = () => {
  const history = useHistory();
  const gotoIssues = (repo: string) => history.push(`/${repo}/issues`);

  return (
    <LinksWrapper>
      {orgs.map((org, index) => {
        const [orgName, repoName] = org.split("/");
        return (
          <div
            key={`${org}-${index}`}
            onClick={() => gotoIssues(org)}
            className="repo"
          >{`go to ${repoName} of ${orgName}`}</div>
        );
      })}
    </LinksWrapper>
  );
};
