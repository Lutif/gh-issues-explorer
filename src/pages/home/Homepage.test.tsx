import { render, screen } from "@testing-library/react";
import { HomePage } from "./HomePage";
import { MemoryRouter } from "react-router-dom";
import { orgs } from "./constansts";

describe("HomePage test", () => {
  it("test HomePage snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders all orgs", () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    orgs.forEach((org) => {
      const [orgName, repoName] = org.split("/");
      const element = `go to ${repoName} of ${orgName}`;
      expect(screen.getByText(element)).toBeInTheDocument();
    });
  });
});
