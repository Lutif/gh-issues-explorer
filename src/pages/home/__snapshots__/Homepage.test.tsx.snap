// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage test test HomePage snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ gncVuK"
  >
    <div
      class="repo"
    >
      go to reactjs.org of reactjs
    </div>
    <div
      class="repo"
    >
      go to material-ui of mui-org
    </div>
    <div
      class="repo"
    >
      go to nuka-carousel of FormidableLabs
    </div>
  </div>
</DocumentFragment>
`;
